### ISSUE TITLE
[java] UnusedPrivateField: false positive when field has the same name as the class

### ISSUE THREAD

**Issue by jsotuyod** [pmd/pmd#1433]
**Affects PMD Version:** 

**Rule:**
`UnusedPrivateField`

**Description:**

False positive when instance variable is an array (or multidimensional array) with the same name as the class. While the code below doesn't follow standard Java naming conventions, it does use the field so `UnusedPrivateField` is irrelevant.

**Code Sample demonstrating the issue:**

```java
public class Board {
    private int[] Board;

    public Board(int[] b) {
        for (int i = 0; i < b.length; i++)
            Board[i] = b[i];
    }

    public int size() {
        return Board.length;
    }
}
```

**Running PMD through:** CLI

**Comment by keerthikorivi** [comment]
The instance variable could be any datatype say int or boolean and not just arrays

**Comment by jsotuyod** [comment]
This is no longer reproducible under PMD 7.0.0
